src/react/useConnectionState.ts

```diff
@@ -0,0 +1,35 @@
+import { useState } from 'react';
```

Comment:
Could we not use the hooks from `ably/react` - `ably` is peer dependency either way?

Reply:
Are you suggesting using ably/react version of hook? We can but I am not sure we should expose this internal hook and some refactoring have to be done on `ably/react` side

Reply:
I think ideally we'd not introduce too much duplication at this stage - you don't want to write tests twice for example, or 3 months later down the line make someone figure out the particular differences (like we did with EventEmitter).

I think also, spaces hooks will almost 100% need to be used in tandem with `ably/react`, I don't think there is much point in trying to separate them.

Reply:
For connection I reused listener from `ably/react`. But unfortunately `ably/react` expects that channel is created, it has no lazy-load mechanism, so until we make `Spaces.get` synchronous we can't use  `ably/react` channels-based hooks right now. I will take a look at both side, `spaces` and `ably/react`. But for now I suggest use duplicate it's small amount of code.